react/components/ProgressBar.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+function ActiveBar () {
+  return <span className="progress-bar progress-bar--active">| </span>;
+}
+
+function InactiveBar () {
+  return <span className="progress-bar progress-bar--inactive">| </span>;
+}
+
+class ProgressBar extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      size       : 12,
+      bars       : [],
+      index      : 0,
+      incrementer: 1,
+    };
+    this.startProgressBar = this.startProgressBar.bind(this);
+    this.updateProgressBar = this.updateProgressBar.bind(this);
+    this.stopProgressBar = this.stopProgressBar.bind(this);
+  }
+  componentWillMount () {
+    const bars = [];
+    for (let i = 0; i <= this.state.size; i++) {
+      bars.push(<InactiveBar key={i} />);
```

Comment:
Instead of adding react elements to state, just store the data.

Then in the `render` function, you can do
`this.state.bars.map((bar, index) => bar.isActive ? <ActiveBar key={index} /> : <InactiveBar key={index} />)`